Models/global_config.hpp

```diff
@@ -0,0 +1,51 @@
+#include <cstdint>
+
+typedef enum {
+    yaw,
+    pitch,
+    roll,
+    thrust
+} ControlAxis_t;
+
+typedef enum {
+    positive,
+    negative
+} ControlDirection_t;
+
+//TODO: Determine correct location for motor driver definitions, and finish implementation.
+class MotorDriver{};
+class TempPWMDriver : public MotorDriver{};
+class TempDSHOTDriver : public MotorDriver{};
+
+typedef struct {
+    ControlAxis_t axis;
+    ControlDirection_t direction;
+    MotorDriver driver;
+    //TODO: Consider a more specific datatype for interfaceID
+    //(e.g. a datatype defined within the motor driver class)
+    uint8_t interfaceID;
```

Comment:
I assume this will ultimately become the constructor arguments for the driver. (e.g. for pwm will be something like timer handle & timer channel). I don't mind the idea of creating this datatype defined in the motor driver class, but we should consider that it could be different for each driver, and so we need a way to vary it based on the driver we are going to use. 

Whatever decision we make here will likely also propagate out to the sensor drivers as well, so it makes sense that we are generic about how we handle this if possible.

Reply:
I defined an interface identifier (IID) for each specific driver that inherits a more generic IID depending on the general type of driver. These definitions would have to be moved to the actual drivers. Outside of config and specific driver classes, the IID's can be referred to using their generic base class, so that knowledge of the specific driver type isn't needed.

Reply:
So I assume the IID would include information that such as the peripheral handle (e.g. `htim1`) and the channel that is used (e.g. `TIM_CHANNEL_2`) and the structure of the IID would be defined by the driver?